honeybee_radiance_folder/folder.py

```diff
@@ -228,425 +354,184 @@ def daylight_matrix_files(self, aperture, receiver=None):
-
-    @property
-    def root(self):
-        """Output root folder."""
-        return self.OUTPUT_ROOT
-
-    @property
-    def octree(self):
-        """Folder for octree files."""
-        return self.OCTREE
-
-    @property
-    def matrix(self):
-        """Matrix outputs folder."""
-        return self.MATRIX
-    
-    @property
-    def temp(self):
-        """Temporary files folder."""
-        return self.TEMP
-    
-    @property
-    def postprocess(self):
-        """Postprocess folder."""
-        return self.postprocess
-
-    def write(self, overwrite=True):
-        """Write model folder.
-
-        Args:
-            overwrite (bool): Set to True to overwrite the folder is it already exist.
+        Returns:
+            path to folder.
         """
-        for subfolder in self.OUTPUT:
-            directory = os.path.join(self.folder, subfolder)
-            if os.path.exists(directory) and not overwrite:
-                raise ValueError('{} already exist.'.format(directory))
-            os.makedirs(directory)
-
+        assert 0 <= folder_type <= 2, 'folder_type must be an integer between 0-2.'
 
-class SystemFolder(_FolderCollection):
-    """System folder includes settings for runs like options, tasks.json."""
-    SYSTEM_ROOT = 'system'
-    SYSTEM = (SYSTEM_ROOT,)
-    REQUIRED = (SYSTEM_ROOT,)
+        if not cfg:
+            cfg = dict(config.minimal)
 
-    @property
-    def root(self):
-        """System root folder."""
-        return self.SYSTEM_ROOT
+        if folder_type == 0:
+            cfg['GRID'] = True
+        elif folder_type == 1:
+            cfg['VIEW'] = True
+        else:
+            cfg['GRID'] = True
+            cfg['VIEW'] = True
 
-    def write(self, overwrite=True):
-        """Write model folder.
+        root_folder = self.model_folder(full=True)
+        if overwrite:
+            shutil.rmtree(root_folder, ignore_errors=True)
 
-        Args:
-            overwrite (bool): Set to True to overwrite the folder is it already exist.
-        """
-        for subfolder in self.SYSTEM:
-            directory = os.path.join(self.folder, subfolder)
-            if os.path.exists(directory) and not overwrite:
+        if not overwrite and os.path.isdir(root_folder):
+            raise ValueError(
+                'Model folder already exist: %s\nSet overwrite to True'
+                ' if you want the folder to be overwritten.' % root_folder
+            )
+        for category in self.CFG_KEYS:
+            if category == 'GLOBAL':
+                continue
+            if not cfg[category]:
+                continue
+            directory = os.path.join(root_folder, self._config[category]['path'])
+            if not os.path.exists(directory):
+                os.makedirs(directory)
+            elif not overwrite:
                 raise ValueError('{} already exist.'.format(directory))
-            os.makedirs(directory)
 
+        shutil.copy2(self._config_file, root_folder)
```

Comment:
Interesting. Is `copy2` better than `copy`?

Reply:
The difference is that the second argument can be a folder instead of a file.

Reply:
Good to know.